src/utils/makeReactNativeConfig.js

```diff
@@ -381,19 +376,21 @@ function injectPolyfillIntoEntry({
   root,
   initializeCoreLocation = 'node_modules/react-native/Libraries/Core/InitializeCore.js',
   dev = true,
+  disableHotReloading = false,
 }: {
   entry: WebpackEntry,
   root: string,
   initializeCoreLocation?: string,
   dev?: boolean,
+  disableHotReloading?: boolean,
 }) {
   const reactNativeHaulEntries = [
     ...getPolyfills(),
     require.resolve(path.join(root, initializeCoreLocation)),
-    require.resolve('./polyfillEnvironment.js'),
   ];
 
-  if (dev) {
+  if (dev && !disableHotReloading) {
+    reactNativeHaulEntries.push(require.resolve('./polyfillEnvironment.js'));
```

Comment:
Why polyfills are only applied in dev now?

Reply:
Correct me if I’m wrong, but it seems like it only sets up the environment for hot reloading. I’m happy to move it back otherwise.

Reply:
Looks like we could require this from hot patch to avoid confusion

Reply:
Good idea. I think we should also rename it while we’re at it. I could do this in a separate PR if you don’t mind. Don’t want to scope creep on this one. 😀

Reply:
Let's do this

Reply:
Thanks! Follow up here: #527